### PR TITLE
Use bufferShim for node 4 support

### PR DESCRIPTION
**Pull request by mifi**
Replace all Buffer.alloc and Buffer.from calls with buffer-shims
Also "npm install" has corrected the order of dependencies

My editor (Atom) has automatically removed indentation on empty lines too

I have tested on node4 and node6 and it solves the issue (#322)

### CODE REVIEW

File: lib/HAPServer.js [hunk 1]
```diff
@@ -317,21 +318,21 @@ HAPServer.prototype._handlePairStepOne = function(request, response, session) {
   var srpParams = srp.params["3072"];
 
   srp.genKey(32, function (error, key) {
-    
+
     // create a new SRP server
-    var srpServer = new srp.Server(srpParams, Buffer.from(salt), Buffer.from("Pair-Setup"), Buffer.from(this.accessoryInfo.pincode), key);
+    var srpServer = new srp.Server(srpParams, bufferShim.from(salt), from("Pair-Setup"), bufferShim.from(this.accessoryInfo.pincode), key);
```
Comment (KhaosT) [inline lib/HAPServer.js]: Please fix ``from("Pair-Setup")``

Comment (mifi) [inline lib/HAPServer.js]: Wow nice catch. Don't know how that happened. I'll fix and take another look over the whole PR too.

**Comment by mifi** [comment]
I reviewed the changes. Looks fine to me now.

**Comment by KhaosT** [comment]
Looks good

**Pull request merged**